parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -1342,7 +1342,16 @@ prettyDoc2 ppe ac tm = case tm of
       (toDocItalic ppe -> Just d) ->
         PP.group $ "*" <> rec d <> "*"
       (toDocBold ppe -> Just d) ->
-        PP.group $ "__" <> rec d <> "__"
+        let inner = rec d
+            numUnderscores =
+              case
+                  filter (\s -> take 2 s == "__")
+                    $ group (PP.toPlainUnbroken $ PP.syntaxToColor inner)
+                of
+                  [] -> 2
+                  x  -> 1 + (maximum $ map length x)
+            underscores = replicate numUnderscores '_'
+         in PP.group $ PP.string underscores <> inner <> PP.string underscores
       (toDocStrikethrough ppe -> Just d) ->
         PP.group $ "~~" <> rec d <> "~~"
```

Comment:
Nice. Do you want to give the same treatment to strikethrough? It'll have the same bug when pretty-printing.

Not sure what to do about italics but we could consider that out of scope for this PR. One idea I had was use 1 or more `*` for italics, and 1 or more `_` for bold. That way you can apply the same trick for italics, bold, and strikethrough.

Reply:
I kind of like that solution, although for some reason I want it to be the other way around (`_` for emphasis, `*` for bold).

Reply:
Sure